google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/mutaterows/MutateRowsAttemptCallable.java

```diff
@@ -250,7 +258,12 @@ private void handleAttemptError(Throwable rpcError) {
     currentRequest = builder.build();
     originalIndexes = newOriginalIndexes;
 
-    throw new MutateRowsException(rpcError, allFailures.build(), entryError.isRetryable());
+    ErrorDetails errorDetails = null;
+    if (rpcError instanceof ApiException) {
+      errorDetails = ((ApiException) rpcError).getErrorDetails();
+    }
+    throw new MutateRowsException(
+        rpcError, allFailures.build(), entryError.isRetryable(), errorDetails);
```

Comment:
You are already passing rpcError to MutateRowsException, why do you need extract the error details first and pass them in parallel?

Reply:
MutateRowsException extends ApiException, which now has an ErrorDetails field. RetryInfoRetryAlgorithm checks .getErrorDetails() to determine if it should use retry delay for error handling. Passing the errorDetails here so I don't need to add special handling for MutateRowsException in RetryInfoRetryAlgorithm, which will be moved to gax later.